### PR TITLE
CRUD of OIDC clients

### PR DESCRIPTION
**Pull request by Jsyro**
Open to changing the model defaults, kind of a janky way to updating the client repository, hopefully it does not cause any issues later.

Stateless is basically what we are doing.
MongoWrapper is incompatible with our version of PyMongo, https://github.com/IdentityPython/pyop/issues/48
RedisWrapper would be a misuse of redis as we would need to add persistence to it.

### CODE REVIEW

File: oidc-controller/api/authSessions/crud.py [hunk 1]
```diff
@@ -43,19 +43,23 @@ async def get(self, auth_session_id: str) -> AuthSession:
 
         return AuthSession(**auth_sess)
 
-    async def patch(self, auth_session_id: str, data: AuthSessionPatch) -> AuthSession:
+    async def patch(self, id: str, data: AuthSessionPatch) -> AuthSession:
```
Comment (esune) [inline oidc-controller/api/authSessions/crud.py:46]: Should we check whether the provided id is valid in other CRUD methods as well other that `post`?

**Review by esune — COMMENTED** [review]
Looks good, only one small comment requiring feedback 🙂 
I am thinking we also may want to add capability to load a config file, but we can tackle that as a separate issue.

File: oidc-controller/api/authSessions/crud.py (hunk 1 above)
Comment (Jsyro) [reply oidc-controller/api/authSessions/crud.py:46]: Ok, i figured out why this is. 

The patch method called by the acapy-webhook handler passes in `id` as a bson.objectid which is a superclass of the PyObjectId that is being used.... but the get method is being passed `id` as a string, so i needed to convert it 

I can leave it and leave a comment as to why this is? or I can cast it to the str in the webhook handler and convert it to a PyObjectId in the CRUD method for consistency. 

I think i'd prefer the second. Thoughts?

Comment (esune) [reply oidc-controller/api/authSessions/crud.py:46]: I agree with the second approach for consistency

**Pull request merged**